[PyTorch] Adopt faster Tuple::create

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65505
* #65504
* __->__ #65381

The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

swolchok committed:
[PyTorch] Adopt faster Tuple::create

The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

ezyang reviewed (approved):
i vetted half the use sites, trusting the rest

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
give up on making TupleElements noncopyable on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Adopt faster Tuple::create"


The previous diff adds a way to make Tuples of size 3 or less
more efficiently. This diff makes it easier to hit that path and
updates a bunch of callsites to hit it.

Differential Revision: [D31069538](https://our.internmc.facebook.com/intern/diff/D31069538/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31069538/)!

[ghstack-poisoned]
